doc/java-generator.md

```diff
@@ -0,0 +1,85 @@
+# Java generation (preview)
+
+This document shows how to generate java sources from CRD manifests.
+
+## Quick start CLI
+
+The Java generator CLI is available for download on Sonatype at the link:
+
+```
+https://oss.sonatype.org/content/repositories/releases/io/fabric8/java-generator-cli/<version>/java-generator-cli-<version>.sh
+```
+
+you can get it working locally with few lines:
+
+```bash
+export VERSION=<version>
+wget https://oss.sonatype.org/content/repositories/releases/io/fabric8/java-generator-cli/$VERSION/java-generator-cli-$VERSION.sh
+chmod a+x java-generator-cli-$VERSION.sh
+./java-generator-cli-$VERSION.sh --version
+```
+
+## Quick start Maven
+
+The Java generator Maven plugin can be used from your project `pom.xml` adding a section like:
+
+```xml
+<plugin>
+  <groupId>io.fabric8</groupId>
+  <artifactId>java-generator-maven-plugin</artifactId>
+  <version><version></version>
+  <executions>
+    <execution>
+      <goals>
+        <goal>generate</goal>
+      </goals>
+    </execution>
+  </executions>
+  <configuration>
+    ...
```

Comment:
it would be nice to have a concrete example of  maven xml configuration with `source` and `target`

Reply:
or is the `source` be only defined with java properties?

Reply:
added `source` as `target` is always optional and already refers to a meaningful default.